### PR TITLE
refactor activity call into consistent convention

### LINKED ISSUE #1576: stateless LFNext calls need to accommodate stateful legacy PHP calls
> This doesn't really address the core issue described in #1536. We will likely run into this issue (wrong project context) as we build out more views in Svelte, so we need to take a step back and engineer something that allows the Svelte code to set the project Id directly. The AngularJS application doesn't have to do this because the project ID is embedded in the URL of the PHP code that serves up the application. The svelte application works differently, so we need to add functionality to our API to facilitate this new way.
> 
> How about adding a PHP API method called `setProject` whose purpose is to set the project context before calling subsequent API methods.
> 
> I think ultimately we want an api where you pass the project id as a parameter (stateless), but that would require re-architecting all the api methods and calling (a big job). We would design our new rest API in this way, with all the things.

I can take a look at converting the project code to an id in the Svelte backend before any legacy calls.  I'd suggest we get this fix in ASAP since that's a bug right now and we can circle back with our desired, long-term solution.  What are your thoughts on that approach?

_Originally posted by @longrunningprocess in https://github.com/sillsdev/web-languageforge/issues/1563#issuecomment-1298709695_

### PR DESCRIPTION
**Pull request by longrunningprocess**
Fixes #1576 

## Description

This PR extends the convention of keeping the stateless calls into the LFNext portions of the app in sync with the stateful requirements of the legacy app.

### Type of Change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

N/A

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

## How to test

- Simply hitting the project dashboard on a project with activity and ensuring there are no failures is a good test.
- While on the dashboard for a project, ensure the "Show all" works as well
- Also ensure clicking on "work on this project" and coming back to the dashboard retains correct project

## qa.languageforge.org testing

Testers should add his/her findings to end of the PR in a comment and include screenshots, files, etc that are beneficial.

### CODE REVIEW

**Review by megahirt — APPROVED** [review]
I approve of this change if this is the way that you want to go.  I do think that moving toward a stateless API surface where you pass the project ID in is a good direction to go in (one we've discussed before), so it's a bit of a shame to undo that in this PR.  But you would have to be willing to refactor the activity_list_dto_for_project() to also do the work that set_project() already does in order to get the same effect.

We have multiple goals in this code, and one of them is keeping the code decoupled, which you are doing.  The other would be to move toward a stateless API.  Obviously we have a ways to go but I leave it up to you in determining when the right time is to refactor some of this - if at all.

**Comment by longrunningprocess** [comment]
in my earlier research, the bigger issue is the `checkPermissions` call, that would have to be refactored to get the context right before checking authz.  It's definitely possible but a bit more intrusive to existing calls if we were to change something there.

**Comment by josephmyers** [comment]
Ran through the bullet points on QA. Looks good

**Pull request merged**
